Client/mods/deathmatch/logic/CClientGif.cpp

```diff
@@ -0,0 +1,356 @@
+/*****************************************************************************
+ *
+ *  PROJECT:     Multi Theft Auto
+ *  LICENSE:     See LICENSE in the top level directory
+ *
+ *  Multi Theft Auto is available from https://www.multitheftauto.com/
+ *
+ *****************************************************************************/
+
+#include "StdInc.h"
+#include "CClientGif.h"
+
+#define GIF_ALLOCATE(pointer,size,release) pointer = (uint8_t*)realloc((release) ? 0 : pointer,(release) ? size : 0UL)
+#define BIGE 0
+#define _SWAP(h) ((BIGE) ? ((uint16_t)(h << 8) | (h << 8)) : h)
+
+CClientGif::CClientGif(CClientManager* pManager, ElementID ID, CGifItem* p_GifItem)
+    : ClassInit(this), CClientTexture(pManager, ID, p_GifItem)
+{
+    SetTypeName("gif");
+    m_pManager = pManager;
+    stride = GetRenderItem()->m_uiSizeX * 4;
+    m_pGifDisplay = std::make_unique<CClientGifDisplay>(m_pManager->GetDisplayManager(), this);
+}
+
+CClientGif::~CClientGif(){
+    Unlink();
+}
+
+void CClientGif::Register(std::vector<unsigned char*> frms, std::vector<int> dls) {
```

Comment:
It's better to pass vectors by a reference or use a move semantics

Reply:
Move would be better here

Reply:
do you mean i should use their address ? for example `CClientGif::Register(std::vector<unsigned char*>& frms)`

Reply:
I mean you should use std::move() to pass these vectors. It should be 
`void CClientGif::Register(std::vector<unsigned char*>&& frms, std::vector<int>&& dls)`

Reply:
fixed!